ern/graphql.go

```diff
@@ -238,160 +505,342 @@ func (g *Resolver) SoundRecording(args soundRecordingArgs) ([]*soundRecordingRes
 	default:
 		return nil, errors.New("Missing ID or Title argument in query")
 	}
-
 	if err != nil {
 		return nil, err
 	}
-
 	defer rows.Close()
 
-	var response []*soundRecordingResolver
-
+	var recordings []*SoundRecordingResolver
 	for rows.Next() {
 		var objectID string
-
 		if err := rows.Scan(&objectID); err != nil {
 			return nil, err
 		}
 
-		id, err := cid.Parse(objectID)
-
+		recording, err := g.soundRecordingResolver(objectID)
 		if err != nil {
 			return nil, err
 		}
 
-		obj, err := g.store.Get(id)
+		recordings = append(recordings, recording)
+	}
+
+	if err := rows.Err(); err != nil {
+		return nil, err
+	}
+
+	return recordings, nil
+}
+
+func (r *Resolver) soundRecordingResolver(objectID string) (*SoundRecordingResolver, error) {
+	id, err := cid.Parse(objectID)
+	if err != nil {
+		return nil, err
+	}
+
+	obj, err := r.store.Get(id)
+	if err != nil {
+		return nil, err
+	}
+
+	source, err := obj.GetString("@source")
+	if err != nil {
+		return nil, err
+	}
 
+	soundRecordingId, err := DecodeObj(r.store, obj, "SoundRecordingId", "ISRC")
+	if err != nil {
+		return nil, err
+	}
+
+	soundRecordingType, err := DecodeObj(r.store, obj, "SoundRecordingType")
+	if err != nil {
+		return nil, err
+	}
+
+	referenceTitle, err := DecodeObj(r.store, obj, "ReferenceTitle", "TitleText")
+	if err != nil {
+		return nil, err
+	}
+
+	recording := &SoundRecordingResolver{
+		resolver:           r,
+		cid:                objectID,
+		source:             source,
+		soundRecordingId:   soundRecordingId.Value,
+		soundRecordingType: soundRecordingType.Value,
+		referenceTitle:     referenceTitle.Value,
+	}
+
+	detailIDs, err := decodeLinks(r.store, obj, "SoundRecordingDetailsByTerritory")
+	if err != nil {
+		return nil, err
+	}
+	for _, detailID := range detailIDs {
+		obj, err := r.store.Get(detailID)
 		if err != nil {
 			return nil, err
 		}
-
-		var ArtistName struct {
-			Value string `json:"@value"`
-		}
-
-		if err := DecodeObj(g.store, obj, &ArtistName, "SoundRecordingDetailsByTerritory", "DisplayArtist", "PartyName", "FullName"); err != nil {
+		titleIDs, err := decodeLinks(r.store, obj, "Title")
+		if err != nil {
 			return nil, err
 		}
-
-		var GenreText struct {
-			Value string `json:"@value"`
+		var titles []*TitleResolver
+		for _, titleID := range titleIDs {
+			obj, err := r.store.Get(titleID)
```

Comment:
can it be obj, err = r.store.Get(titleID) ?
obj is already been declared

Reply:
`obj` is only used within this `for` loop, so I don't see why it needs to change?